src/main/java/net/snowflake/ingest/utils/HttpUtil.java

```diff
@@ -42,9 +44,27 @@ public class HttpUtil {
   private static String PROXY_SCHEME = "http";
   private static int MAX_RETRIES = 3;
 
-  private static HttpClient httpClient;
+  private static CloseableHttpClient httpClient;
 
-  public static HttpClient getHttpClient() {
+  private static PoolingHttpClientConnectionManager connectionManager;
+
+  private static final int DEFAULT_CONNECTION_TIMEOUT_MINUTES = 1;
+  private static final int DEFAULT_HTTP_CLIENT_SOCKET_TIMEOUT_MINUTES = 5;
+
+  // Default is 2, but scaling it up to 100 to match with default_max_connections
+  private static final int DEFAULT_MAX_CONNECTIONS_PER_ROUTE = 100;
+
+  // 100 is close to max partition number we have seen for a kafka topic ingesting into snowflake.
+  private static final int DEFAULT_MAX_CONNECTIONS = 100;
```

Comment:
This is depends on the number of tasks but not partitions, right?

Reply:
This is per partition since we create SimpleINgestManager per partition.